### DISCUSSION TITLE
How to use "light.glsl" and "screen.glsl".

### DISCUSSION THREAD

**Discussion by xiaocats** [bnpr/Malt#65] [Q&A] [answered]
I know that ”light.glsl” is a shader used in the light section, but what effect can it actually have?
(When I apply "basic.light.glsl", the screen is still dark)
And for "screen.glsl", I don't know where to use it.
I've recently started learning GLSL, but I'm not yet proficient enough to understand these codes just by reading them. Sorry about that.

**Comment by pragma37** [comment, marked as answer]
https://github.com/bnpr/Malt/tree/master/Malt/Pipelines/NPR_Pipeline#pipeline-shaders

Light shaders let you specify the color/intensity contribution of a light at any given surface point.
Think for example a projector, a bonfire, or any stylized lighting effects.

For the basic light example to work you need to set a texture, otherwise you are just projecting a black (ie. no intensity) light.

Screen shaders are used for post-processing or any other kind of screen-space technique.
To use them you need to declare and use them from a custom pipeline. You can select a custom pipeline in the world settings.

For testing, you can use the CustomPipeline.py from the example folders. After loading, there will be a Custom Post Process parameter where you can use a screen shader material.

**Reply by xiaocats** [reply]
Thank you!
Thanks to you I know how to use it a little better, but I think I'll have a hard time getting used to it.
I'm going to try to learn it little by little.
